code/datums/components/climb_walkable.dm

```diff
@@ -0,0 +1,29 @@
+/// Allows objects that entered parent's tile to move freely through other objects with this component regardless of density
+/datum/component/climb_walkable
```

Comment:
Should this be an element? I can imagine us spawning a lot of these for some maps or ruins, and nothing here seems particularly bespoke

Reply:
We can't use an element because we are using `connect_loc_behalf`, which in turn means that this component is "holding state" (just in a way that is slightly harder to notice).   
Each individual instance of the component needs its own instance of the other component.